highlight/highlight
# fix grouping of non mapped stacktrace errors

## Summary



This PR ensures that backend only errors have a fingerprint set. The previous code only applies to errors with `mapped_stack_trace` set (i.e. does it have a source map).

When we added backend only errors, we missed confirming if fingerprinting works. As a result, errors with the same stacktrace (but different error bodies) are not grouped together:

![Screenshot 2023-05-15 at 9 56 32 AM](https://github.com/highlight/highlight/assets/58678/99b80a49-6689-4f38-bfb9-3f40bfc3e5f2)


Finally, this PR does a slew of organization to make the code more readable (see inline comments).

Fixes #5240



## How did you test this change?



* Create a backend only error
* Ensure that a fingerprint exists (see #5240)

See this backend error with fingerprinting
![Screenshot 2023-05-15 at 10 04 32 AM](https://github.com/highlight/highlight/assets/58678/f0537c82-3904-44af-813b-bef81dda524a)

See also this demo that does a full clicktest:
https://www.loom.com/share/44162631603b450381c3792cc8252b09


## Are there any deployment considerations?



N/A
